x-pack/plugins/upgrade_assistant/server/lib/reindexing/worker.ts

```diff
@@ -111,6 +117,9 @@ export class ReindexWorker {
 
         // Push each operation through the state machine and refresh.
         await Promise.all(this.inProgressOps.map(this.processNextStep));
+        // TODO: This tight loop needs something to relax potentially high CPU demands so this padding is added.
+        // This scheduler should be revisited in future.
+        await new Promise(res => setTimeout(res, WORKER_PADDING_MS));
```

Comment:
Should we only pause/delay if are no operations that we have credentials for?

Reply:
Good point, yes we should!